net/qpid-proton/Portfile

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.0
+
+name                qpid-proton
+description         A Qpid Proton description goes here
+long_description \
+    Qpid Proton is a high-performance, lightweight messaging library. It can \
+    be used in the widest range of messaging applications, including brokers, \
+    client libraries, routers, bridges, proxies, and more. Proton makes it \
+    trivial to integrate with the AMQP 1.0 ecosystem from any platform, \
+    environment, or language.
+version             0.19.0
+categories          net
+license             Apache-2
+maintainers         @roddiekieley
+platforms           darwin freebsd linux
+homepage            https://qpid.apache.org
+
+fetch.type          git
+git.url             git://github.com/apache/qpid-proton.git
+git.branch          fa80534c87940b44fefd36abff4c5a9a79b99f47
+
+cmake.out_of_source yes
+
+configure.args-append \
+                    -DCMAKE_INSTALL_RPATH:STRING="${prefix}/lib" \
+                    -DCMAKE_INSTALL_NAME_DIR:STRING="${prefix}/lib" \
+                    -DSASL_IMPL=none \
+                    -DSSL_IMPL=none \
+                    -DLIB_SUFFIX=""
+
+default_variants +swig +openssl
+
+variant openssl description {With built-in support for OpenSSL} {
+    configure.args-replace -DSSL_IMPL=none -DSSL_IMPL=openssl
+    depends_lib     port:openssl
+}
+
+variant swig description {With built-in support for SWIG so the bindings can be built} {
+    depends_lib     port:swig \
```

Comment:
Same as above.

I'm additionally worried how exactly this varian works. Will the port be opportunistically linking against these ports when present? Is there a way to prevent enabling swing when `swig-*` ports are present?

Reply:
@mojca Thank you for reviewing this item and all the commentary above, will address.

For this particular aspect the CMakeLists.txt in the project are set up to detect swig and utilize it if necessary. As well it detects if the various language implementations are available. I could pass through appropriate cmake defines to disable each of the bindings. 

Perhaps that's a safer approach, and then I can make an update later to add the swig variant back to get the rest of the language bindings that require it?

Reply:
You should pass a variable that disables all swig bindings when the variant is not enabled. When the variant is enabled, you need to replace those flags. (There's no need to create a zillion variants, just one variant that enables all the bindings is fine. All that's missing are a few extra flags that prevent opportunistic bindings.)